test/evp_xof_test.c

```diff
@@ -0,0 +1,455 @@
+/*
```

Comment:
Could this be done using evp_test?

More specifically, why not add XOF functionality to evp_test?

I prefer to see data driven tests where possible.

Reply:
I think we should have both. This is a quite specialized test and I do not think all these various ways to examine the functionality is needed in evp_test. But making it possible to call the squeeze multiple times in evp_test would be nice.

Reply:
Er, evp_test does have XOF functionality, mind...

Reply:
Only via the DigestFinalXOF call.  That is single shot output.

Reply:
Ah, right.  I'd suggest making that a separate effort.

Reply:
@levitte, does this mean adding it to evp_test or not?

Reply:
I think this is not part of this PR.